include/aos/common/pkcs11/pkcs11.hpp

```diff
@@ -0,0 +1,711 @@
+    /**
+     * Attribute value.
+     */
+    Array<uint8_t> mValue;
+};
+
+/**
+ * Incapsulates session-related routines of Cryptoki API.
+ */
+class SessionContext {
+public:
+    /**
+     * Returns session information.
+     *
+     * @param[out] info result information.
+     * @return Error.
+     */
+    Error GetSessionInfo(SessionInfo& info) const
+    {
+        (void)info;
+
+        return {};
+    }
+
+    /**
+     * Logs a user into a token.
+     *
+     * @param userType type of Cryptoki user.
+     * @param pin user’s PIN.
+     * @return Error.
+     */
+    Error Login(UserType userType, const String& pin)
+    {
+        (void)userType;
+        (void)pin;
+
+        return {};
+    }
+
+    /**
+     * Logs a user out from a token.
+     *
+     * @return Error.
+     */
+    Error Logout() { return {}; }
+
+    /**
+     * Initializes the normal user's PIN.
+     *
+     * @param pin user pin.
+     * @return Error.
+     */
+    Error InitPIN(const String& pin)
+    {
+        (void)pin;
+
+        return {};
+    }
+
+    /**
+     * Obtains an attribute values of the specified object.
+     *
+     * @param handle object handle.
+     * @param types types of attributes to be returned.
+     * @param[out] values result attribute values.
+     * @return Error.
+     */
+    Error GetAttributeValues(ObjectHandle handle, const Array<AttributeType>& types, Array<Array<uint8_t>>& values)
+    {
+        (void)handle;
+        (void)types;
+        (void)values;
+
+        return {};
+    }
+
+    /**
+     * Searches for token and session objects that match a template.
+     *
+     * @param templ template for search operation.
+     * @param[out] objects result object handles.
+     * @return Error.
+     */
+    Error FindObjects(const Array<ObjectAttribute>& templ, Array<ObjectHandle>& objects)
+    {
+        (void)templ;
+        (void)objects;
+
+        return {};
+    }
+
+    /**
+     * Creates a new object.
+     *
+     * @param templ object’s template.
+     * @return RetWithError<ObjectHandle>.
+     */
+    RetWithError<ObjectHandle> CreateObject(const Array<ObjectAttribute>& templ)
+    {
+        (void)templ;
+
+        return {0U, Error::Enum::eNone};
+    }
+
+    /**
+     * Destroys object by its handle.
+     *
+     * @param handle handle of the object to be destroyed.
+     * @return Error.
+     */
+    Error DestroyObject(ObjectHandle handle)
+    {
+        (void)handle;
+
+        return {};
+    }
+
+    /**
+     * Returns session handle.
+     *
+     * @return session handle.
+     */
+    SessionHandle GetHandle() const { return {}; }
+
+    /**
+     * Destroyd object instance.
+     */
+    virtual ~SessionContext() = default;
+};
+
+/**
+ * PKCS11 Context instance.
+ */
+class LibraryContext {
+public:
+    /**
+     * Initializes PKCS11 library.
+     *
+     * @param library underlying PKCS11 C library path
+     * @return Error
+     */
+    Error InitLibrary(const String& library)
```

Comment:
Take into account we may have different modules that use the same pkcs11 library. Need to discuss.